pkg/engine/engine.go

```diff
@@ -1016,25 +1016,27 @@ func (cad *cadEngine) UpdatePackageResources(ctx context.Context, repositoryObj
 	resources := repository.PackageResources{
 		Contents: prevResources.Spec.Resources,
 	}
-	appliedResources, _, err := applyResourceMutations(ctx, draft, resources, mutations)
+
+	appliedResources, renderStatus, err := applyResourceMutations(ctx, draft, resources, mutations)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	// render the package
-	// Render failure will not fail the overall API operation.
-	// The render error and result is captured as part of renderStatus above
-	// and is returned in packageresourceresources API's status field. We continue with
-	// saving the non-rendered resources to avoid losing user's changes.
-	// and supress this err.
-	_, renderStatus, _ := applyResourceMutations(ctx,
-		draft,
-		appliedResources,
-		[]mutation{&renderPackageMutation{
-			runnerOptions: runnerOptions,
-			runtime:       cad.runtime,
-		}})
-
+	if len(appliedResources.Contents) > 0 {
```

Comment:
Not sure if this is the right solution. The applyResourceMutations is executed with a single mutator, called defined at engine.go#1007. I'd think if the old and the new resources are the same, it should return with a full list instead of an empty one. 

Or the idea is that this is how we differentiate between a no-op update, and an update that requires a re-render?